amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShardConsumer.java

```diff
@@ -95,9 +96,10 @@
     private final InternalSubscriber subscriber;
 
     public ShardConsumer(RecordsPublisher recordsPublisher, ExecutorService executorService, ShardInfo shardInfo,
-                         Optional<Long> logWarningForTaskAfterMillis, ShardConsumerArgument shardConsumerArgument) {
+                         Optional<Long> logWarningForTaskAfterMillis, ShardConsumerArgument shardConsumerArgument,
+                         TaskExecutionListener taskExecutionListener) {
         this(recordsPublisher, executorService, shardInfo, logWarningForTaskAfterMillis, shardConsumerArgument,
-                ConsumerStates.INITIAL_STATE,
+                taskExecutionListener, ConsumerStates.INITIAL_STATE,
```

Comment:
Is there a reason for changing the order for the chained constructor? If not could you add the new parameter to the end of the list.

Reply:
Mainly, I just wanted to avoid adding a reference type (TaskExecutionListener) after a primitive type (int bufferSize) in the constructor as it didn't look aesthetically pleasing.
But I'm fine with adding this new argument to the end of the list.
Why would you not want to use builder pattern for the constructor here?